### DISCUSSION TITLE
What is the maximum number of local processes (consumers) for Sidekiq?

### DISCUSSION THREAD

**Discussion by georgeu2000** [sidekiq/sidekiq#6488] [Q&A] [answered]
I want to run a single instance of Sidekiq on AWS with the largest number of CPU's available, 896.

There would be 894 processes consuming Sidekiq jobs. What is the upper limit for number of consumers? Does this limit change with the license?

**Comment by mperham** [comment]
There's no hard coded limits. I've heard of customers running Redis with thousands of Sidekiq threads. Sidekiq Enterprise's multi-process support might be essential to save a ton of memory. Can you share anything more about your use case, either publicly here or privately via email?

**Reply by georgeu2000** [reply]
Thanks for the quick reply.

Sure, it's a data mining project. Each process only runs one thread (for simplicity). Each process only requires appox. 300MB, so memory is not a limiting factor (896 vCPU instances start at 12TB of memory). Currently runs fine with 8 processes on my dev machine. Each job takes about 13 min to complete, so the demand on Sidekiq is actually minimal.

I choose to run on a single high CPU machine for simplicity; I don't need to manage multiple instances, set up virtual networking, a separate instance to run Sidekiq, etc. It's just for a personal project.

**Comment by mperham** [comment, marked as answer]
It should work fine, especially with a local Redis.

**Reply by georgeu2000** [reply]
Got it. Thanks for all your help.
